src/main/java/org/opentripplanner/service/vehiclerental/street/Composite.java

```diff
@@ -0,0 +1,97 @@
+package org.opentripplanner.service.vehiclerental.street;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.opentripplanner.street.model.RentalRestrictionExtension;
+import org.opentripplanner.street.search.state.State;
+
+/**
+ * Combines multiple restrictions into one.
+ */
+public final class Composite implements RentalRestrictionExtension {
+
+  private final RentalRestrictionExtension[] exts;
+
+  private Composite(RentalRestrictionExtension... exts) {
+    for (var ext : exts) {
+      if (ext instanceof org.opentripplanner.service.vehiclerental.street.Composite) {
+        throw new IllegalArgumentException(
+          "Composite extension cannot be nested into one another."
+        );
+      }
+    }
+    var set = new HashSet<>(Arrays.asList(exts));
+    this.exts = set.toArray(RentalRestrictionExtension[]::new);
+  }
+
+  @Override
+  public boolean traversalBanned(State state) {
+    for (var ext : exts) {
+      if (ext.traversalBanned(state)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @Override
+  public boolean dropOffBanned(State state) {
+    for (var ext : exts) {
+      if (ext.dropOffBanned(state)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @Override
+  public Set<RestrictionType> debugTypes() {
+    var set = new HashSet<RestrictionType>();
+    for (var e : exts) {
```

Comment:
You used `ext` in the other loops. In Java (and some other languges) `it` is commonly used as the current element in loops (iteration). I prefer `it` here and above, but you may choose either `it` or `ext`  - your choice.